### ISSUE TITLE
The default behaviour of `put` is not sane enough for unknown object types

### ISSUE THREAD

**Issue by augustebaum** [probabl-ai/skore#822]
### Is your feature request related to a problem? Please describe.

The current default when an unknown object type is `put` is to raise an `ItemTypeError`.

The user has two solutions if they want to store their object in skore:

- Use `MediaItem`, i.e. rather than:
```
project.put("key", my_obj)
```
write for example
```
project.put_item("key", MediaItem.factory(my_obj_as_bytes_or_str, media_type="text/html"))
```
This means the user has the responsibility of passing the media type and to convert their object to bytes or string form, which they are unlikely to do in practice.
- Open an issue to ask for a new `Item` type (or open a PR adding it). See #806 for example.

Both of these solutions lower the usability of skore.

### Describe the solution you'd like

We should support `put` for any object that has a `_repr_html_` method, and when this is the case we do the conversion to `MediaItem` ourselves.

In the future we can also consider other media types by detecting other `_repr_xxx_` methods, e.g. by looking first for `_repr_html_`, then `_repr_svg_`, then `_repr_png_`, etc. We might also make use of `_repr_mimebundle_`.

### Describe alternatives you've considered, if relevant

Use pickle (not environment-dependent, but could be more useful for users)

### Additional context

This issue is derived from https://github.com/probabl-ai/skore/pull/810#discussion_r1858639637

**Comment by augustebaum** [comment]
@MarieS-WiMLDS This doesn't seem ready to me

**Comment by MarieS-WiMLDS** [comment]
You describe the solution you want, do why isn't it ready?

**Comment by rouk1** [comment]
I vote that we do this. This a refactor as it means that we will have to write a presenter for each Item type and remove the serialization related stuff from the API routes. This is a first step to have our widgets usable in notebook. See related PR #926 .

**Comment by rouk1** [comment]
> @rouk1 I'm not sure I get what you mean. This issue simply says: "upon `put`, if the object matches none of our Item types, then try saving `object._repr_html_()`"

You're right I did not red the issue summary correctly. My comment is irrelevant.

**Comment by tuscland** [comment]
This will be closed by #1045

**Comment by thomass-dev** [comment]
No longer relevant after #1045 .
